src/Dev/Deprecation.php

```diff
@@ -146,12 +151,18 @@ protected static function get_called_method_from_trace($backtrace, $level = 1)
         if (!$level) {
             $level = 1;
         }
+        $called = Deprecation::get_called_from_trace($backtrace, $level);
+        return ($called['class'] ?? '') . ($called['type'] ?? '') . ($called['function'] ?? '');
+    }
+
+    private static function get_called_from_trace(array $backtrace, int $level): array
+    {
         $newLevel = $level;
-        // handle closures inside withSuppressedNotice()
-        if (Deprecation::$insideNoticeSuppression
-            && substr($backtrace[$newLevel]['function'], -strlen('{closure}')) === '{closure}'
-        ) {
-            $newLevel = $newLevel + 2;
+        if (Deprecation::$insideNoticeSuppression) {
+            // handle closures inside withSuppressedNotice()
+            if (substr($backtrace[$newLevel]['function'], -strlen('{closure}')) === '{closure}') {
+                $newLevel = $newLevel + 2;
+            }
```

Comment:
I prefer this how it was (an if that only contains another if is a bit of a code smell) - though that's subjective so I won't hold back merging if you leave it as-is.

I will note it took me a good 10 minutes to try to figure out what changed here before I realised it was only unrelated stylistic changes.

Reply:
Updated